openedx/core/djangoapps/programs/tests/test_views.py

```diff
@@ -0,0 +1,214 @@
+                    'category': 'xseries',
+                    'status': 'active',
+                    'course_codes': [
+                        {
+                            'organization': {'display_name': 'Test Organization 1', 'key': 'edX'},
+                            'marketing_slug': 'fake-marketing-slug-xseries-1',
+                            'display_name': 'Demo XSeries Program 1',
+                            'key': 'TEST_A',
+                            'run_modes': [
+                                {'sku': '', 'mode_slug': 'ABC_1', 'course_key': 'edX/DemoX_1/Run_1'},
+                                {'sku': '', 'mode_slug': 'ABC_2', 'course_key': 'edX/DemoX_2/Run_2'},
+                            ]
+                        }
+                    ],
+                    'subtitle': 'Dummy program 1 for testing',
+                    'name': 'First Program'
+                },
+            }
+            self.assertTrue(mock_get.called)
+            self.assertEqual(expected_output, programs)
+            self.assertEqual(sorted(programs.keys()), ['edX/DemoX_1/Run_1', 'edX/DemoX_2/Run_2'])
+
+            # now test with user having multiple courses across two different
+            # xseries
+            mock_get.reset_mock()
+            programs = get_course_programs_for_dashboard(
+                self.user,
+                ['edX/DemoX_1/Run_1', 'edX/DemoX_2/Run_2', 'edX/Program/Program_Run', 'valid/edX/Course']
+            )
+            expected_output['edX/Program/Program_Run'] = {
+                'category': 'xseries',
+                'status': 'active',
+                'course_codes': [
+                    {
+                        'organization': {'display_name': 'Test Organization 2', 'key': 'edX'},
+                        'marketing_slug': 'fake-marketing-slug-xseries-2',
+                        'display_name': 'Demo XSeries Program 2',
+                        'key': 'TEST_B',
+                        'run_modes': [
+                            {'sku': '', 'mode_slug': 'XYZ_1', 'course_key': 'edX/Program/Program_Run'},
+                        ]
+                    }
+                ],
+                'subtitle': 'Dummy program 2 for testing',
+                'name': 'Second Program'
+            }
+            self.assertTrue(mock_get.called)
+            self.assertEqual(expected_output, programs)
+            self.assertEqual(
+                sorted(programs.keys()),
+                ['edX/DemoX_1/Run_1', 'edX/DemoX_2/Run_2', 'edX/Program/Program_Run']
+            )
+
+    def test_get_course_programs_with_exception(self):
+        """ Test that the method 'get_course_programs_for_dashboard' returns
+        empty dictionary in case of exceptions.
+        """
+        # mock the request call
+        with patch('slumber.Resource.get') as mock_get:
+            # test output in case of the exception 'ValueError'
+            mock_get.side_effect = ValueError('exc')
```

Comment:
I would be comfortable with a single test here for `Exception` instead of one each for (`ValueError`, `ImproperlyConfigured`, `HttpClientError`).  However, I think we should also have a test where we mock an exception coming from patching `programs_api_client` (or `slumber.API.__init__`, either way is equivalent I think) - this will exercise the other place in which we are catching exceptions in that function.